src/core/processing/qgsprocessingparameters.cpp

```diff
@@ -4013,6 +4091,42 @@ QgsProcessingParameterGeometry *QgsProcessingParameterGeometry::fromScriptCode(
   return new QgsProcessingParameterGeometry( name, description, definition, isOptional );
 }
 
+QString QgsProcessingParameterGeometry::userFriendlyString( const QVariant &value ) const
+{
+  if ( value.isValid() )
+  {
+
+    if ( value.userType() == qMetaTypeId< QgsGeometry>() )
+    {
+      const QgsGeometry g = value.value<QgsGeometry>();
+      return g.typeName();
+    }
+
+    else if ( value.userType() == qMetaTypeId<QgsReferencedGeometry>() )
+    {
+      const QgsReferencedGeometry g = value.value<QgsReferencedGeometry>();
+      if ( !g.isNull() )
+      {
+        return QStringLiteral( "[%1] %2" ).arg( g.crs().userFriendlyIdentifier( Qgis::CrsIdentifierType::ShortString ), g.typeName() );
```

Comment:
let's flip this order -- the geometry type should come first, then the crs (to match eg extent and point parameter formats)